config/routes.rb

```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+devise_for :users, controllers: { registrations: 'users/registrations' }, sign_out_via: [:get, :delete]
+   root "welcome#index"
+    resources :groups do
+    resources :expenses
```

Comment:
- [ ] It will be great to only generate the routes that you need in your application. For example, you are not using the `delete routes` or the `edit routes` anywhere in your application. So the best thing to do is to use `only` or `expect` to generate only the needed routes in your application. To learn more about this, kindly see this [article](https://guides.rubyonrails.org/routing.html#shallow-nesting)